x/datapool/types/pool.go

```diff
@@ -44,8 +41,35 @@ func GetModuleAddress() sdk.AccAddress {
 	return authtypes.NewModuleAddress(ModuleName)
 }
 
-func GetAccumPoolShareToken(poolID, amount uint64) sdk.Coin {
-	return sdk.NewCoin(fmt.Sprintf(ShareTokenPrefix+"/%v", poolID), sdk.NewIntFromUint64(amount))
+// AppendPoolID adds the poolID.
+func (d *InstantRevenueDistribute) AppendPoolID(poolID uint64) {
+	// Check duplicate
+	for _, existPoolID := range d.PoolIds {
+		if existPoolID == poolID {
+			return
+		}
+	}
+	d.PoolIds = append(d.PoolIds, poolID)
+}
+
+func (d *InstantRevenueDistribute) IsEmpty() bool {
+	return d.PoolIds == nil || len(d.PoolIds) == 0
+}
+
+func (d *InstantRevenueDistribute) RemovePreviousIndex(idx int) {
```

Comment:
I would recommend to rename this to `TruncateFromBeginning(count int)`. I didn't get that the `RemovePreviousIndex` means before reading this function body.

Reply:
OK.
I fixed it.
https://github.com/medibloc/panacea-core/pull/318/commits/b398ff659c3333c09e78ccf4def88f4dd20dc9f2